microsoft365/onedrive/pkg/commands/addPermission.go

```diff
@@ -0,0 +1,42 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/client"
+	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/global"
+	"github.com/obot-platform/tools/microsoft365/onedrive/pkg/graph"
+)
+
+func AddPermission(ctx context.Context, driveID string, itemID string, emails string, role string, message string, password string, expirationDateTime string) error {
+	if driveID == "me" {
+		return fmt.Errorf("drive_id must be the actual drive ID, cannot be 'me'")
+	}
+
+	var emailList []string
+	if emails != "" {
+		emailList = strings.Split(emails, ",")
```

Comment:
It's probably worth trimming any whitespace off of each of these emails.

Reply:
updated, although very less likely it will ever occur